scene/resources/environment.cpp

```diff
@@ -1211,267 +1308,49 @@ void Environment::_bind_methods() {
-
 	BIND_ENUM_CONSTANT(SSAO_BLUR_DISABLED);
 	BIND_ENUM_CONSTANT(SSAO_BLUR_1x1);
 	BIND_ENUM_CONSTANT(SSAO_BLUR_2x2);
 	BIND_ENUM_CONSTANT(SSAO_BLUR_3x3);
+
+	BIND_ENUM_CONSTANT(SDFGI_CASCADES_4);
+	BIND_ENUM_CONSTANT(SDFGI_CASCADES_6);
+	BIND_ENUM_CONSTANT(SDFGI_CASCADES_8);
+
+	BIND_ENUM_CONSTANT(SDFGI_Y_SCALE_DISABLED);
+	BIND_ENUM_CONSTANT(SDFGI_Y_SCALE_75_PERCENT);
+	BIND_ENUM_CONSTANT(SDFGI_Y_SCALE_50_PERCENT);
 }
 
 Environment::Environment() {
 	environment = RS::get_singleton()->environment_create();
 
-	bg_mode = BG_CLEAR_COLOR;
-	bg_sky_custom_fov = 0;
-	bg_energy = 1.0;
-	bg_canvas_max_layer = 0;
-	ambient_energy = 1.0;
-	//ambient_sky_contribution = 1.0;
-	ambient_source = AMBIENT_SOURCE_BG;
-	reflection_source = REFLECTION_SOURCE_BG;
-	set_ambient_light_sky_contribution(1.0);
-	set_camera_feed_id(1);
-
-	tone_mapper = TONE_MAPPER_LINEAR;
-	tonemap_exposure = 1.0;
-	tonemap_white = 1.0;
-	tonemap_auto_exposure = false;
-	tonemap_auto_exposure_max = 8;
-	tonemap_auto_exposure_min = 0.05;
-	tonemap_auto_exposure_speed = 0.5;
-	tonemap_auto_exposure_grey = 0.4;
-
-	set_tonemapper(tone_mapper); //update
-
-	adjustment_enabled = false;
-	adjustment_contrast = 1.0;
-	adjustment_saturation = 1.0;
-	adjustment_brightness = 1.0;
-
-	set_adjustment_enable(adjustment_enabled); //update
-
-	ssr_enabled = false;
-	ssr_max_steps = 64;
-	ssr_fade_in = 0.15;
-	ssr_fade_out = 2.0;
-	ssr_depth_tolerance = 0.2;
-
-	ssao_enabled = false;
-	ssao_radius = 1;
-	ssao_intensity = 1;
-	ssao_bias = 0.01;
-	ssao_direct_light_affect = 0.0;
-	ssao_ao_channel_affect = 0.0;
-	ssao_blur = SSAO_BLUR_3x3;
-	set_ssao_edge_sharpness(4);
-
-	glow_enabled = false;
-	glow_levels = (1 << 2) | (1 << 4);
-	glow_intensity = 0.8;
-	glow_strength = 1.0;
-	glow_mix = 0.05;
-	glow_bloom = 0.0;
-	glow_blend_mode = GLOW_BLEND_MODE_SOFTLIGHT;
-	glow_hdr_bleed_threshold = 1.0;
-	glow_hdr_luminance_cap = 12.0;
-	glow_hdr_bleed_scale = 2.0;
-
-	fog_enabled = false;
-	fog_color = Color(0.5, 0.5, 0.5);
-	fog_sun_color = Color(0.8, 0.8, 0.0);
-	fog_sun_amount = 0;
-
-	fog_depth_enabled = true;
-
-	fog_depth_begin = 10;
-	fog_depth_end = 100;
-	fog_depth_curve = 1;
-
-	fog_transmit_enabled = false;
-	fog_transmit_curve = 1;
-
-	fog_height_enabled = false;
-	fog_height_min = 10;
-	fog_height_max = 0;
-	fog_height_curve = 1;
-
-	set_fog_color(Color(0.5, 0.6, 0.7));
-	set_fog_sun_color(Color(1.0, 0.9, 0.7));
-
-	sdfgi_enabled = false;
-	sdfgi_cascades = SDFGI_CASCADES_6;
-	sdfgi_min_cell_size = 0.2;
-	sdfgi_use_occlusion = false;
-	sdfgi_use_multibounce = false;
-	sdfgi_read_sky_light = false;
-	sdfgi_enhance_ssr = false;
-	sdfgi_energy = 1.0;
-	sdfgi_normal_bias = 1.1;
-	sdfgi_probe_bias = 1.1;
-	sdfgi_y_scale = SDFGI_Y_SCALE_DISABLED;
+	set_camera_feed_id(bg_camera_feed_id);
+	_update_ambient_light();
+	_update_tonemap();
+	_update_ssr();
+	_update_ssao();
 	_update_sdfgi();
-}
+	_update_glow();
+	_update_fog();
+	_update_fog_depth();
+	_update_fog_height();
+	_update_adjustment();
```

Comment:
Some setters were called here instead of only setting the member variables, presumably to force calling updates in `RenderingServer` and propagate the default config.

That's why I've changed it to now call `_update_*` methods directly and eventually a `_change_notify()` as some methods like `set_adjustment_enable()` would have triggered.

There's however still the properties from the Background/Sky category which don't get their `RS` equivalent called here (but also didn't before), with the exception of `set_camera_feed_id()` which I kept. Should I add calls for all RS::bg methods too?